newIDE/app/src/AssetStore/CreditsPackages/CreditsPackagePurchaseDialog.js

```diff
@@ -0,0 +1,346 @@
+    () => {
+      if (initialCreditsAmount !== null && limits && isPurchasing) {
+        const newCreditsAmount = limits.credits.userBalance.amount;
+        if (newCreditsAmount > initialCreditsAmount) {
+          setIsPurchasing(false);
+          setPurchaseSuccessful(true);
+        }
+      }
+    },
+    [initialCreditsAmount, isPurchasing, purchaseSuccessful, limits]
+  );
+
+  const dialogContents = !profile
+    ? {
+        subtitle: <Trans>Log-in to purchase these credits</Trans>,
+        content: (
+          <CreateProfile
+            onOpenLoginDialog={onOpenLoginDialog}
+            onOpenCreateAccountDialog={onOpenCreateAccountDialog}
+            message={
+              <Trans>
+                Credits will be linked to your user account. Log-in or sign-up
+                to purchase them!
+              </Trans>
+            }
+            justifyContent="center"
+          />
+        ),
+      }
+    : purchaseSuccessful
+    ? {
+        subtitle: <Trans>Your purchase has been processed!</Trans>,
+        content: (
+          <Line justifyContent="center" alignItems="center">
+            <Text>
+              <Trans>You can now use them across the app!</Trans>
+            </Text>
+          </Line>
+        ),
+      }
+    : isPurchasing
+    ? {
+        subtitle: shouldUseOrSimulateAppStoreProduct ? (
+          <Trans>Complete your purchase with the app store.</Trans>
+        ) : (
+          <Trans>Complete your payment on the web browser</Trans>
+        ),
+        content: shouldUseOrSimulateAppStoreProduct ? (
+          <>
+            <ColumnStackLayout justifyContent="center" alignItems="center">
+              <CircularProgress size={40} />
+              <Text>
+                <Trans>
+                  The purchase will be linked to your account once done.
+                </Trans>
+              </Text>
+            </ColumnStackLayout>
+          </>
+        ) : (
+          <>
+            <Line justifyContent="center" alignItems="center">
+              <CircularProgress size={20} />
+              <Spacer />
+              <Text>
+                <Trans>Waiting for the purchase confirmation...</Trans>
+              </Text>
+            </Line>
+            <Spacer />
+            <Line justifyContent="center">
+              <BackgroundText>
+                <Trans>
+                  Once you're done, come back to GDevelop and the credits will
+                  be added to your account automatically.
+                </Trans>
+              </BackgroundText>
+            </Line>
+          </>
+        ),
+      }
+    : !limits
+    ? {
+        subtitle: <Trans>Loading your profile...</Trans>,
+        content: (
+          <Line justifyContent="center" alignItems="center">
+            <CircularProgress size={20} />
+          </Line>
+        ),
+      }
+    : {
+        subtitle: (
+          <Trans>
+            {creditsPackageListingData.name} will be added to your account{' '}
```

Comment:
I think you can remove the `{' '}`. It may be easier to translate on crowdin

Reply:
Oh yeah, it was added automatically so thought it was needed but looks like not